poetry/utils/env.py

```diff
@@ -677,8 +682,20 @@ def build_venv(cls, path, executable=None):
 
         build(path)
 
-    def remove_venv(self, path):  # type: (str) -> None
-        shutil.rmtree(path)
+    @classmethod
+    def remove_venv(cls, path, keep_venv_dir=False):  # type: (str) -> None
+        if not keep_venv_dir:
+            shutil.rmtree(path)
+            return
+
+        # Otherwise, delete all files and folders but the toplevel one
+        for filename in os.listdir(path):
+            file_path = os.path.join(path, filename)
+
+            if os.path.isfile(file_path) or os.path.islink(file_path):
+                os.remove(file_path)
+            elif os.path.isdir(file_path):
+                shutil.rmtree(file_path)
```

Comment:
Hello @TheButlah,

thanks a lot for contribution! :+1: 

I think we can simplify this, by always deleting the content of the venv folder instead of removing the folder. We should leave a comment why we don't delete the whole folder, otherwise there is a good chance the code get reverted at some time.

To be consistent with the rest of the code, could you please use `Path` instead of `os.path`?

fin swimmer

Reply:
Hey @finswimmer!

Sure, I can get the code to use `Path` instead of `os.path`!

When I added the option to either delete or keep the virtualenv folder, I wasn't sure under which any situations the folder iteself should be deleted. For example, I noticed that poetry seems to randomly (?) generate a string for the venv -would there be a situation where the venv gets renamed due to this or any other reason? Because if so ostensibly poetry should delete the old env along with its folder right?

Reply:
@finswimmer just following on my question. Is it actually safe to always leave the folder? I would expect that it should only be left when `in-project` is set to `True`

Reply:
Hello @TheButlah, the name for the venv folder isn't random. It's hash based on the python version used, the package name and the path. Recreating is only triggered when poetry detects, that the venv folder it wants to use, is broken. So I think deleting the content of the folder and not the folder directly is always possible.

Reply:
@finswimmer OK, sounds good. I made `remove_venv` simply empty the folder contents in all scenarios. Requesting a review.